team-builder/src/components/Form.jsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState, useEffect } from "react";
+import "../../src/styles.css";
+
+// id for array
+let idCounter = () => {
+  let id = 0;
+  return function countUp() {
+    return id++;
+  };
+};
+
+const arrId = idCounter();
+
+const Form = props => {
+  const [member, setMember] = useState({ name: "", email: "", role: "" });
+
+  const handleChange = e => {
+    setMember({ ...member, [e.target.name]: e.target.value });
+  };
+
+  useEffect(() => {
+    setMember(props.memberToEdit);
+  }, [props.memberToEdit]);
+
+  const handleSubmit = e => {
+    e.preventDefault();
+    // props.setTeamMembers([...props.teamMembers, member]);
+    // setMember({ name: "", email: "", role: "" });
+    if (!props.isEditing) {
+      props.setTeamMembers([...props.teamMembers, { ...member, id: arrId() }]);
+      props.setisEditing(false);
+      setMember({ name: "", email: "", role: "" });
+    } else if (props.isEditing) {
+      const updatedList = props.teamMembers.filter(
+        teamMember => teamMember.id === props.memberToEdit
```

Comment:
This is what's causing your bug:
```suggestion
        teamMember => teamMember !== props.memberToEdit
```
So, now it will return all of the members that are not equal to the member you want to edit.